fuel-core/src/tx_pool.rs

```diff
@@ -83,9 +85,43 @@ impl TxPool {
 
     pub async fn submit_tx(&self, tx: Transaction) -> Result<Bytes32, Error> {
         let db = self.db.clone();
-        let tx_id = tx.id();
+
+        let mut tx_to_exec = tx.clone();
+
+        let includable_txs: Vec<Transaction>;
+
+        if self.executor.config.utxo_validation {
+            if tx_to_exec.metadata().is_none() {
+                tx_to_exec.precompute_metadata();
+            }
+
+            self.fuel_txpool
+                .insert(vec![Arc::new(tx_to_exec.clone())])
+                .await
+                .into_iter()
+                .collect::<Result<Vec<_>, _>>()?;
+
+            let includable_arc_txs = self.fuel_txpool.includable().await;
+
+            includable_txs = includable_arc_txs
+                .iter()
+                .map(|arc| Transaction::clone(&*arc))
+                .collect();
+
+            for included_tx in includable_arc_txs {
+                self.fuel_txpool.remove(&[included_tx.id()]).await;
+            }
+        } else {
+            includable_txs = vec![tx];
+        }
+
+        let tx_id = tx_to_exec.id();
+
         // set status to submitted
-        db.update_tx_status(&tx_id, TransactionStatus::Submitted { time: Utc::now() })?;
+        db.update_tx_status(
+            &tx_id.clone(),
+            TransactionStatus::Submitted { time: Utc::now() },
+        )?;
```

Comment:
@Voxelot  this seems like it needs a rework, updating status before it is executed is not a good flow, everything before block/tx inclusion shouldn't leave any trace in database, only after block is included we could have status of it. 
Expected status of transaction is:
* Pending (inside txpool)
* Included (included in chain)

Is sdk using this information?

Reply:
I agree this needs rework and this was called out on the original [task](https://github.com/FuelLabs/fuel-core/issues/152):
> Update transaction status API to fallback to the txpool if the status doesn't exist in the database, using the Submitted status

I'm assuming there will be follow-up PR's to finish the rest of the planned work? @ControlCplusControlV

Reply:
@Voxelot yes I am planning on using a fallback based on tx status rather than existence in a separate PR, as I figured there may be some overlap with #183 in part atleast, by tackling

> if the transaction was very invalid (i.e. not includable in a block) this seems like it might need to be separate type of error from a tx that was partially processed within a block (i.e. utxos were spent and outputs created, but the script panicked). 

As I could generate the first status based on if it errors when inserting into the tx pool (which would require utxo verification to be on to determine if a tx is invalid) and then the second would be a result of a failure but the tx was includable.

Unless you would rather that separate PR just cover lookup in mempool by status?